logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java

```diff
@@ -45,7 +48,9 @@ public interface PipelineBus {
     static PipelineBus create() {
         final String pipelineBusImplementation = System.getProperty("logstash.pipelinebus.implementation", "v2");
         switch (pipelineBusImplementation) {
-            case "v1": return new PipelineBusV1();
+            case "v1":
+                DEPRECATION_LOGGER.deprecated("The legacy pipeline bus selected with `logstash.pipelinebus.implementation=v1` is deprecated, and will be removed in a future release of Logstash");
+                return new PipelineBusV1();
```

Comment:
It looks like in the case we have an "unknown" configuration setting we "default" to v1 still. Maybe as part of calling out v1 is deprecated we can also change that fallback to also be v2. 

```diff
diff --git a/logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java b/logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java
index 487793d62..b10fa6729 100644
--- a/logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java
+++ b/logstash-core/src/main/java/org/logstash/plugins/pipeline/PipelineBus.java
@@ -54,7 +54,7 @@ public interface PipelineBus {
             case "v2": return new PipelineBusV2();
             default:
                 LOGGER.warn("unknown pipeline-bus implementation: {}", pipelineBusImplementation);
-                return new PipelineBusV1();
+                return new PipelineBusV2();
         }
     }
```